ui/js/search.js

```diff
@@ -0,0 +1,171 @@
+        }
+        return false;
+    });
+
+
+
+});
+
+function displayDropdown(dataArr) {
+    var currentlySelected = -1;
+    var numOptions = 0;
+    var dropdownOpen = false;
+    var disableArrowSelection = false;
+
+    initDialog(dataArr);
+
+    function initDialog(courses) {
+        numOptions = courses.length;
+        $('.dialog').empty();
+        for (var course of courses) {
+            if (chosenCourses.indexOf(course.id) < 0) {
+                $('.dialog').append('<div data-course-id=' + course.id + '>' +
+                    course.courseFullName.replaceAll(":", "") + '</div>');
+            }
+        }
+        $('.dialog').addClass('open');
+        dropdownOpen = true;
+        resetEventListeners(); 
+    }
+
+    function trackArrowKeysAndEnterBtn() {
+        $('.autocomplete').keydown(function (e) {
+            e.stopPropagation();
+
+            switch (e.which) {
+                case 13: // enter button 
+                    chooseOptionWithElement($('.dialog > div').eq(currentlySelected));
+                    break;
+                case 38: // up
+                    selectOptionWithIndex(-1);
+                    break;
+                case 40: // down
+                    selectOptionWithIndex(1);
+                    break;
+                default: return;
+            }
+        });
+    }
+
+    function selectOptionWithIndex(arrowDirection) {
+        if ((currentlySelected < 1 && arrowDirection < 1) ||
+            (currentlySelected === numOptions - 1 && arrowDirection > -1) ||
+            !dropdownOpen || disableArrowSelection) {
+            return;
+        }
+        var options = $('.dialog > div');
+        var currentSelection = options.eq(currentlySelected);
+        currentSelection.removeClass('selected');
+        currentSelection.removeAttr('id');
+        currentlySelected += arrowDirection;
+        startSelectOptionVisuals(options, arrowDirection);
+    }
+
+    function startSelectOptionVisuals(searchResults, arrowDirection) {
+        var newSelection = searchResults.eq(currentlySelected);
+        newSelection.addClass('selected');
+        newSelection.attr('id', 'selectedItem');
+
+        var scrollDivisor = 1.20;
+        if (arrowDirection == -1) {
+            scrollDivisor = 2;
+        }
+        $('.dialog').animate({
+            scrollTop: $("#selectedItem").offset().top / scrollDivisor
+        }, 100);
+        $('.autocomplete input').val(newSelection[0].innerHTML);
+        $('.autocomplete input').css('color', '#757575');
+    }
+
+    function toggleSearchResultsDialog(forceClose) {
+        $('.dialog > div').eq(currentlySelected).removeAttr("selectedItem");
+        if (forceClose) {
+            $('.dialog').removeClass('open');
+            dropdownOpen = false;
+        } else {
+            $('.dialog').toggleClass('open');
+            dropdownOpen = !dropdownOpen;
+        }
+    }
+
+    function chooseOptionWithElement(element) {
```

Comment:
Can we move this and `setDeleteCourseListener` to a function outside the `displayDropdown` function? Since these actions aren't part of the `displayDropdown` process?